src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -173,6 +174,36 @@ public function loadClass($class)
                 @trigger_error(sprintf('The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.', $parent, self::$final[$parent], $name), E_USER_DEPRECATED);
             }
 
+            // Only check final methods if the class has a parent
+            if ($parent) {
+                foreach ($refl->getMethods(\ReflectionMethod::IS_PUBLIC | \ReflectionMethod::IS_PROTECTED) as $method) {
+                    try {
+                        $prototype = $method->getPrototype();
```

Comment:
parsing the docblock (doing an strpos preflight really) should be faster than trying getPrototype, then fall in the catch block, which involves instaciating a ReflectionException behind the scene - especially since this situation is going to happen quite often (since many methods don't have "protototype"